app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2SampleProvider.java

```diff
@@ -47,13 +47,23 @@
     // 12 = definitely wake up
     // 17 = definitely not sleep related
 
+    // Firmware >1.0.1.47 changed sleep detection
+    // Those activities are marked wit NEWFW
+    // 112 (with intensity >0) - Light sleep
+    // 122 - Deep sleep
+    // 28 - waking up + walking in the same minute
+    // 105 - waking up but staying in bed (no steps but high raw intensity)
+    // 123 - start of sleep
+
     public static final int TYPE_UNSET = -1;
     public static final int TYPE_NO_CHANGE = 0;
     public static final int TYPE_ACTIVITY = 1;
     public static final int TYPE_NONWEAR = 3;
     public static final int TYPE_CHARGING = 6;
     public static final int TYPE_LIGHT_SLEEP = 9;
+    public static final int TYPE_LIGHT_SLEEP_NEWFW = 112;
```

Comment:
When was the type changed? 1.0.1.47?
Can you rename the variables to ..._1-0-1-47?

Reply:
No please don't. This code is shared with Amazfit Bip at least right now, and there firmware versions are different.

Reply:
@ashimokawa Well its another device. It may have its own file in my opinion.

Reply:
@Roxxor91 
Sure, if we find out that the devices are really diverging.

Up to a certain firmware both acted the same, and now it seems differen. Amazfit Bip is only one month old, so it could also be a bug that it acts up.
Since both use the same Mifit app I think they will stay very similar - but we will see.

Reply:
@ashimokawa Well you can join the analysis in Issue #686 and do it really detailed. For that you need to sync your amazfit with MiFit and Gadgetbridge. MiFit shows exact times when you were in which Phase, which helps a lot to try to recognize patterns.
I do not think that the Amazfit is acting up. Mifit supports the old MiBands and the MiBand 2 which already differ by now.

Reply:
Are we sure we're not getting a "record version" in some unknown byte from the miband (and the bip)? Pebble watches send back such an information and it allows us to properly decode the data.

Reply:
FWIW, I really appreciate the work done here, thanks a lot! :+1: 

To be really correct, we should check the firmware version that was used to produce the sample. Otherwise we might mix up values that have meaning X in version 1.0 and meaning Y in 1.2.

I will see how we can do this in an efficient way.